src/Microsoft.Diagnostics.Tools.RuntimeClient/DiagnosticsIpc/IpcMessage.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.Diagnostics.Tools.RuntimeClient.DiagnosticsIpc
+{
+    public class IpcMessage
+    {
+        public IpcMessage()
+        { }
+
+        public IpcMessage(IpcHeader header, byte[] payload)
+        {
+            Payload = payload;
+            Header = header;
+        }
+
+        internal IpcMessage(DiagnosticServerCommandSet commandSet, byte commandId, byte[] payload = null)
+        {
+            Header = new IpcHeader(commandSet, commandId);
+            Payload = payload;
+        }
+
+        public byte[] Payload { get; private set; } = null;
+        public IpcHeader Header { get; private set; } = default;
+
+        public byte[] Serialize()
+        { 
+            byte[] serializedData = null;
+            UInt16 size = IpcHeader.HeaderSizeInBytes;
+
+            // Verify things will fit in the size capacity
+            try
+            {
+                size = checked((UInt16)(IpcHeader.HeaderSizeInBytes + Payload.Length));
+            }
+            catch (OverflowException e)
+            {
+                // TODO error checking
+                throw e;
```

Comment:
Throw is probably the right way of handling this error, we can simply delete the whole try-catch block and let it throw at the right spot.